### PR TITLE
Feature/#548 new email editor page

### PR DESCRIPTION
**Pull request by Sinharitik589**
## Description

Provide feature of editing the email templates providing user to set subject as well as body for the template 
Major changes/addition:-

- Seperate mongodb model for email  as setting model usecase doesnt apply to it , it has additional field like subject and moreover we dont want to fetch email configurations along with all settings as it is an important aspect
- An additional key emailVariables assigned to the model , for the usecase when we need to place the variables for e.g. name as per user which can now be achieved as user will use the format {{variable}} for those cases.

## Related Issues
#548

## Screenshots (if applicable)
ezgif com-video-to-gif (16)

## Checklist

- [x] I have tested these changes
- [ ] I have updated the relevant documentation
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the codebase
- [x] My changes generate no new warnings or errors
- [x] The title of my pull request is clear and descriptive

### CODE REVIEW

**Comment by salahlalami** [comment]
@Sinharitik589 Cool , it's look good , i like your approach to create new Modal for Email

File: models/coreModels/Email.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  emailVariables:{
+    type:Array
+  },
+  emailBody: {
+    type: String,
+    required: true,
+  },
+  emailSubject: {
+    type: mongoose.Schema.Types.Mixed,
```
Comment (salahlalami) [inline models/coreModels/Email.js:24]: what's reason to use mongoose.Schema.Types.Mixed,
type: String, is could be okay

File: models/coreModels/Email.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    lowercase: true,
+    required: true,
+  },
+  emailName: {
+    type: String,
+    unique:true,
+    required:true,
+  },
+  emailVariables:{
+    type:Array
```
Comment (salahlalami) [inline models/coreModels/Email.js:17]: what's reason to use this emailVariables ?

for custom email variable , we can use pattern {{client.name}} , then when we send email , we can replace those variable before sending mail

File: models/coreModels/Email.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+mongoose.Promise = global.Promise;
+
+const emailSchema = new mongoose.Schema({
+  emailKey: {
+    type: String,
+    unique: true,
+    lowercase: true,
+    required: true,
+  },
```
Comment (salahlalami) [inline models/coreModels/Email.js:10]: Add new Field language : 
type : string
default: en

File: models/coreModels/Email.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+mongoose.Promise = global.Promise;
+
+const emailSchema = new mongoose.Schema({
```
Comment (salahlalami) [inline models/coreModels/Email.js:4]: add default fields :

 removed: {
    type: Boolean,
    default: false,
  },
  enabled: {
    type: Boolean,
    default: true,
  },
  created: {
    type: Date,
    default: Date.now,
  },
 updated: {
    type: Date,
    default: Date.now,
  },

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: models/coreModels/Email.js (hunk 2 above)
Comment (Sinharitik589) [reply models/coreModels/Email.js:17]: This is for two purpose , first is to let user know that these are the varables that you might use in that particular email and second  is to parse only those variables when sending the mail 
image

Comment (salahlalami) [reply models/coreModels/Email.js:17]: perfect

**Review by salahlalami — APPROVED** [review]

**Comment by salahlalami** [comment]
Thank you @Sinharitik589

**Pull request merged**